selfdrive/modeld/modeld.cc

```diff
@@ -101,90 +152,20 @@ int main(int argc, char **argv) {
   LOGW("models loaded, modeld starting");
 
   VisionIpcClient vipc_client = VisionIpcClient("camerad", VISION_STREAM_YUV_BACK, true, device_id, context);
-
-  while (!do_exit){
-    if (!vipc_client.connect(false)){
-      util::sleep_for(100);
-      continue;
-    }
-    break;
+  while (!do_exit && !vipc_client.connect(false)) {
+    util::sleep_for(100);
   }
 
-  // loop
-  while (!do_exit) {
-    VisionBuf *b = &vipc_client.buffers[0];
+  // run the models
+  if (vipc_client.connected) {
```

Comment:
Can this ever be not true?

Reply:
vipc_client.connected is false only when do_exit is true